src/lib.rs

```diff
@@ -82,6 +82,19 @@ impl Config {
                 wait_time: None,
             })
     }
+
+    pub fn new_from_consul_host(consul_host: &str) -> Result<Config> {
```

Comment:
Suggestion, could we also add another port `Option<u16>` to specify the port (and use 8500 if the value is not given)?

Reply:
Good idea. I thought about it at first, but was too lazy :sweat_smile: 
Here it is. Thanks for the fast response.

Reply:
@tomers : A small last one, do you mind also adding an Option<str> for the token?
So, please might use it with restricted token access as well...

Thanks a lot for your contribution!

Reply:
Sure, np. Done

Reply:
@tomers thanks! It seems there is a compilation issue however, checks are failing, do you need help on it?

Reply:
@pierresouchay it seems these failures are not related to my PR. I don't want to introduce unrelated changes here.

Reply:
Indeed, thanks a lot